### PR TITLE
Added tests for kmu and ikg functionality

### PR DESCRIPTION
**Pull request by degjorva**
Added tests for writing to and using keys from the kmu. Added tests for using the ikg to derive a key and signing.

### CODE REVIEW

**Review by nordicjm — COMMENTED** [review]
linewrap commit message to 70 characters per line

File: tests/psa_crypto/tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
```
Comment (nordicjm) [inline tests/psa_crypto/tests/CMakeLists.txt:2]: updates dates

File: tests/psa_crypto/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+tests:
+  psa_crypto:
+    sysbuild: true
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
+
```
Comment (nordicjm) [inline tests/psa_crypto/testcase.yaml:6]:
```suggestion
```

File: tests/psa_crypto/tests/test_ikg_key_derivation.c [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+#endif /* CONFIG_BUILD_WITH_TFM */
+
+/* ====================================================================== */
+/*			Global variables/defines for the ikg key derivation test	  */
+
+#define KEYSLOT HUK_KEYSLOT_MKEK
+#define ENCRYPT_ALG PSA_ALG_GCM
+#define IV_LEN 12
+#define NRF_CRYPTO_TEST_AES_MAX_TEXT_SIZE (52)
+#define NRF_CRYPTO_TEST_AES_ADDITIONAL_SIZE (36)
```
Comment (Vge0rge) [inline tests/psa_crypto/tests/test_ikg_key_derivation.c:33]: This is unused I think.

File: tests/psa_crypto/tests/test_kmu_use.c [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	key_handle = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(
+		CRACEN_KMU_KEY_USAGE_SCHEME_RAW, KMU_SLOT_NUM);
+
+	psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+
+	if (psa_get_key_attributes(key_handle, &attr) != PSA_SUCCESS) {
+		return APP_ERROR;
+
```
Comment (Vge0rge) [inline tests/psa_crypto/tests/test_kmu_use.c:59]:
```suggestion
```
Nit

File: tests/psa_crypto/tests/test_ikg_key_derivation.c [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	status = encrypt_message();
+	TEST_VECTOR_ASSERT_EQUAL(0, status);
+
+	status = decrypt_message();
+	TEST_VECTOR_ASSERT_EQUAL(0, status);
+
+	status = crypto_finish();
+	TEST_VECTOR_ASSERT_EQUAL(0, status);
+
+	if (status != APP_SUCCESS) {
```
Comment (Vge0rge) [inline tests/psa_crypto/tests/test_ikg_key_derivation.c:169]: Same as before, you can just do return status here.

**Review by Vge0rge — COMMENTED** [review]
Very good to have this, overall it looks good! Some minor comments here and there.

File: tests/psa_crypto/testcase.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+tests:
+  crypto.cracen:
+    sysbuild: true
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
```
[the remark this answers is not in the document]
Comment (degjorva) [reply tests/psa_crypto/testcase.yaml:5]: Yes. Updated

File: tests/psa_crypto/prj.conf [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+# Enable logging
+CONFIG_CONSOLE=y
+CONFIG_LOG=n
+
+# Enable nordic security backend and PSA APIs
+CONFIG_NRF_SECURITY=y
+CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+CONFIG_PSA_CRYPTO_DRIVER_CRACEN=y
+CONFIG_PSA_CRYPTO_DRIVER_OBERON=n
```
Comment (frkv) [inline tests/psa_crypto/prj.conf:22]: NIT: Likely not needed
```suggestion
```

File: tests/psa_crypto/tests/test_ikg_identity_key_sign_verify.c [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+LOG_MODULE_DECLARE(app, LOG_LEVEL_DBG);
+
+int get_identity_key(void)
+{
+	psa_status_t status;
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+	size_t data_length;
+
+	key_handle = mbedtls_svc_key_id_make(0, identity_key_id);
```
Comment (frkv) [inline tests/psa_crypto/tests/test_ikg_identity_key_sign_verify.c:57]: Note to self: Ask Mbed TLS team why this uses an inline and not a macro

File: tests/psa_crypto/tests/test_ikg_key_derivation.c [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+LOG_MODULE_DECLARE(app, LOG_LEVEL_DBG);
+
+psa_status_t derive_key(psa_key_attributes_t *attributes, uint8_t *key_label,
+			uint32_t label_size, psa_key_id_t *key_id_out)
+{
+	uint8_t key_out[PSA_BITS_TO_BYTES(128)];
+	psa_status_t status;
+	*key_id_out = PSA_KEY_ID_NULL;
+
+	#if defined(CONFIG_BUILD_WITH_TFM)
```
Comment (frkv) [inline tests/psa_crypto/tests/test_ikg_key_derivation.c:59]: Don't indent, the #if defined here. It is difficult to see the #else/#endif

File: tests/psa_crypto/tests/test_ikg_key_derivation.c [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+	}
+
+	status = psa_import_key(attributes, key_out, sizeof(key_out), &key_id);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	*key_id_out = key_id;
+
+	#endif "CONFIG_BUILD_WITH_TFM"
```
Comment (frkv) [inline tests/psa_crypto/tests/test_ikg_key_derivation.c:113]:
```suggestion
	#endif /* CONFIG_BUILD_WITH_TFM */
```

File: tests/psa_crypto/tests/test_ikg_key_derivation.c [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+	status = psa_key_derivation_input_key(&op, PSA_KEY_DERIVATION_INPUT_SECRET, huk_key_id);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+	status = psa_key_derivation_input_bytes(&op, PSA_KEY_DERIVATION_INPUT_LABEL, key_label,
+						   label_size);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+	status = psa_key_derivation_input_bytes(&op, PSA_KEY_DERIVATION_INPUT_CONTEXT, NULL,
```
Comment (frkv) [inline tests/psa_crypto/tests/test_ikg_key_derivation.c:82]:
```suggestion
	/* Context is optional to use, so it is set to NULL with zero length */
	status = psa_key_derivation_input_bytes(&op, PSA_KEY_DERIVATION_INPUT_CONTEXT, NULL,
```

File: tests/psa_crypto/tests/test_kmu_use.c [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+	psa_status_t status;
+
+	status = get_eddsa_pub_key();
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	/* Verify the signature of the message */
+	status = psa_verify_message(key_handle,
+					PSA_ALG_PURE_EDDSA,
```
Comment (frkv) [inline tests/psa_crypto/tests/test_kmu_use.c:80]: Intentation follows 4-char tab

File: tests/psa_crypto/tests/test_ikg_identity_key_sign_verify.c [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+{
+	psa_status_t status;
+
+	status = psa_verify_message(pub_key_id,
+			PSA_ALG_ECDSA(PSA_ALG_SHA_256),
+			m_plain_text,
+			sizeof(m_plain_text),
+			m_signature,
+			NRF_CRYPTO_TEST_IKG_SIGNATURE_SIZE);
+	if (status != PSA_SUCCESS) {
```
Comment (frkv) [inline tests/psa_crypto/tests/test_ikg_identity_key_sign_verify.c:125]: Newline before helps readabilty

File: tests/psa_crypto/src/psa_tests_common.c [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+
+ZTEST_SUITE(test_suite_ikg, NULL, NULL, NULL, NULL, NULL);
+
+static psa_key_id_t key_id;
+
+int crypto_init(void)
+{
+	psa_status_t status;
+
+	#if !defined(CONFIG_BUILD_WITH_TFM)
```
Comment (frkv) [inline tests/psa_crypto/src/psa_tests_common.c:44]: Indentation unneeded, here

**Review by frkv — APPROVED** [review]
Overall looks good (only formatting comments)

File: tests/psa_crypto/src/psa_tests_common.c [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+		return status;
+	}
+	return APP_SUCCESS;
+}
+
+int crypto_finish(void)
+{
+	psa_status_t status;
+
+	status = psa_destroy_key(key_id);
```
Comment (Vge0rge) [inline tests/psa_crypto/src/psa_tests_common.c:63]: The key_id here is not initialized, right?

Comment (degjorva) [reply tests/psa_crypto/src/psa_tests_common.c:63]: It was, but in an awkward. Moved it into relevant files.

**Review by Vge0rge — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**